net/httpd.c

```diff
@@ -82,10 +84,23 @@ int do_http_upgrade(const ulong size, const int upgrade_type){
         OLED_Clear_page(6);
         OLED_ShowString(0,2,"DownLoad:100%");
 #endif
-        ret = raspi_erase_write((unsigned char*)(WEBFAILSAFE_UPLOAD_RAM_ADDRESS),WEBFAILSAFE_UPLOAD_KERNEL_ADDRESS,size);
+
+		unsigned int upgrade_flash_offset;
+		
+		if (0x5A5A5A5A == sys_bootflag) {
+			upgrade_flash_offset = 0x50000;
```

Comment:
We should change all hardcoded value here to macro define

Reply:
Of course. This is just a rushed design for quick testing